pkg/pillar/utils/file/file.go

```diff
@@ -92,3 +95,27 @@ func StatAndRead(log *base.LogObject, filename string, maxReadSize int) (string,
 	}
 	return string(content[0:n]), fi.ModTime(), err
 }
+
+// ReadSavedCounter returns counter value if the file exists
+// else returns 0
+func ReadSavedCounter(log *base.LogObject, fileName string) uint32 {
+	if log != nil {
+		log.Tracef("ReadSavedCounter - reading %s", fileName)
+	}
+
+	b, _, err := StatAndRead(log, fileName, maxCounterReadSize)
+	if err == nil {
+		c, err := strconv.Atoi(b)
+		if err != nil {
+			if log != nil {
+				log.Errorf("ReadSavedCounter: %s", err)
+				return 0
+			}
+		}
+		return uint32(c)
+	}
+	if log != nil {
+		log.Warnf("ReadSavedCounter - %s doesn't exist", fileName)
```

Comment:
The error could be something else so logging the error makes sense.

Reply:
Reasonable, done.